assets/js/modules/ads/datastore/settings.js

```diff
@@ -38,12 +39,38 @@ export const INVARIANT_INVALID_CONVERSION_ID =
 	'a valid conversionID is required to submit changes';
 
 export async function submitChanges( { select, dispatch } ) {
+	const haveSettingsChanged = select( MODULES_ADS ).haveSettingsChanged();
+	const haveConversionTrackingSettingsChanged =
+		select( CORE_SITE ).haveConversionTrackingSettingsChanged();
+
 	// This action shouldn't be called if settings haven't changed,
 	// but this prevents errors in tests.
-	if ( select( MODULES_ADS ).haveSettingsChanged() ) {
-		const { error } = await dispatch( MODULES_ADS ).saveSettings();
-		if ( error ) {
-			return { error };
+	if ( haveSettingsChanged || haveConversionTrackingSettingsChanged ) {
```

Comment:
Same as in the analytics settings. No need to have multi-level nesting here. Just return early when necessary.